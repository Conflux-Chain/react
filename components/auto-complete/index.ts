import AutoComplete, { useAutoCompleteHandle } from './auto-complete'
export { useAutoCompleteHandle }
export default AutoComplete
