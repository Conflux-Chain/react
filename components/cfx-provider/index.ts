import CfxProvider from './cfx-provider'

export default CfxProvider
