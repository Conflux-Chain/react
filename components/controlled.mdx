If you think the data flow in react carefully.

uncontrolled component:
caller do not know the internal state of the component

controlled component:
the state of called component is fully controlled by `props` passed in 

The above two are identified by react offical team. However, there is also something 
in between, which can not fit in the patterns mentioned above

third patten: the called need to know the callee state sometims but not always
think form as an example, the caller only need to know the state of callee when 'submit'
happen rather than keep track of the component all the time

forth pattern: the caller need to set the callee state forcelly on a one time, not inter
