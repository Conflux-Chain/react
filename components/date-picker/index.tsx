import { Moment } from 'moment'
import momentGenerateConfig from '@jnoodle/rc-picker/lib/generate/moment'
import generatePicker, {
  PickerProps,
  PickerDateProps,
  RangePickerProps as BaseRangePickerProps,
} from './generate-picker'

export type DatePickerProps = PickerProps<Moment>
export type MonthPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type WeekPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type RangePickerProps = BaseRangePickerProps<Moment>

export const DatePicker = generatePicker<Moment>(momentGenerateConfig)

export default DatePicker
