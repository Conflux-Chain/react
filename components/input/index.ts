import Input, { useInputHandle } from './input'

export { useInputHandle }
export default Input
