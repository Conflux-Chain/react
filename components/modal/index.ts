import Modal from './modal'
import ModalTitle from './modal-title'
import ModalSubtitle from './modal-subtitle'
import ModalContent from './modal-content'
import ModalAction from './modal-action'
import useModalHandle from './use-imperative-modal'

Modal.Title = ModalTitle
Modal.Subtitle = ModalSubtitle
Modal.Content = ModalContent
Modal.Action = ModalAction
Modal.useModalHandle = useModalHandle

export default Modal
