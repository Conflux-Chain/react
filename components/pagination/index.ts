import Pagination from './pagination'
import PaginationPrevious from './pagination-previous'
import PaginationNext from './pagination-next'
import usePaginationHandle from './use-pagination-handle'
Pagination.Previous = PaginationPrevious
Pagination.Next = PaginationNext
Pagination.usePaginationHandle = usePaginationHandle
export default Pagination
