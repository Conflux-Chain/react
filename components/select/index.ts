import Select from './select'
import SelectOption from './select-option'
import useSelectHandle from './use-imperactive-select'

Select.Option = SelectOption
Select.useSelectHandle = useSelectHandle

export default Select
