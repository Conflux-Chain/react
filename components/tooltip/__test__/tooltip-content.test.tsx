import React from 'react'
import { mount } from 'enzyme'
import { CfxProvider } from 'components'
import TooltipContent from '../tooltip-content'

describe('tooltip-content', () => {
  it("should return null when there's no parent", () => {
    const wrapper = mount(
      <CfxProvider theme={{ type: 'dark' }}>
        <TooltipContent color="default" visible hideArrow={false} offset={0} placement="top" />
      </CfxProvider>,
    )
    expect(wrapper.html).toMatchSnapshot()
  })
})
