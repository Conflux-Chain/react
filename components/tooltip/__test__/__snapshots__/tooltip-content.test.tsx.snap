// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip-content should return null when there's no parent 1`] = `[Function]`;
